### PR TITLE
Fix target_folder

### PR DESCRIPTION
**Pull request by devngc**
This PR expands the capability to write to a target folder to the rest of the writers. Without this, the script was writing the vtk files at the current working directory and the .zip file at the target_folder.

### CODE REVIEW

File: honeybee_vtk/write.py [hunk 1]
```diff
@@ -102,21 +110,22 @@ def write_vtk(file_path, *, file_name=None, target_folder=None, include_grids=Tr
 
     # remove extension if provided by user
     file_name = file_name if not file_name.lower().endswith('.zip') else file_name[:-4]
-
-    target_folder = target_folder or os.getcwd()
-    zip_file = os.path.join(target_folder, file_name + '.zip')
```
Comment (mostaphaRoudsari) [inline honeybee_vtk/write.py:107]: why are you removing the check for None target folder?

File: honeybee_vtk/write.py [hunk 2]
```diff
@@ -66,6 +66,13 @@ def write_vtk(file_path, *, file_name=None, target_folder=None, include_grids=Tr
             raise ValueError(writer_error)
     else:
         raise ValueError(writer_error)
+    
+    # Set the target folder to write the files
+    target_folder = target_folder or os.getcwd()
+    if os.path.exists(target_folder):
+        pass
+    else:
+        raise OSError('Path to target_folder is not valid.')
```
[the remark this answers is not in the document]
Comment (devngc) [reply honeybee_vtk/write.py:75]: But what if the folder is there and the user made a mistake in writing path?

File: honeybee_vtk/write.py (hunk 1 above)
Comment (devngc) [reply honeybee_vtk/write.py:107]: I did not remove it. Just shifted it [here](https://github.com/ladybug-tools/honeybee-vtk/pull/9/files#diff-74d6244911d9a54aabb679036a0eb51ae1bc14166f91efe71c6e55e2415b942bR71)

File: honeybee_vtk/writers.py [hunk 3]
```diff
@@ -113,26 +123,29 @@ def _write_grids(grids, vtk_writer, vtk_extension):
     # If base_geometry is found
     if grids[0]:
         base_geo_points = get_grid_base(grids[0])[0]
-        write_polydata(base_geo_points, 'grid base', vtk_writer, vtk_extension)
+        args = [base_geo_points, 'grid base', vtk_writer, vtk_extension, target_folder]
+        write_polydata(*args)
```
[the remark this answers is not in the document]
Comment (devngc) [reply honeybee_vtk/writers.py:127]: Complied

File: honeybee_vtk/write.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (devngc) [reply honeybee_vtk/write.py]: Complied

File: honeybee_vtk/write.py [hunk 4]
```diff
@@ -66,6 +66,17 @@ def write_vtk(file_path, *, file_name=None, target_folder=None, include_grids=Tr
...
+    # Set the target folder to write the files
+    target_folder = target_folder or os.getcwd()
+    if os.path.exists(target_folder):
+        pass
+    else:
+        warnings.warn(
+        'The path provided at target_folder does not lead to a folder.'
+        ' Hence, a new folder is created at the path provided.'
+        )
+        os.makedirs(target_folder, exist_ok=True)
```
Comment (mostaphaRoudsari) [inline honeybee_vtk/write.py:79]: This is more elegant if you use `not`.

```suggestion
    if not os.path.exists(target_folder):
        warnings.warn(
        'The path provided at target_folder does not lead to a folder.'
        ' Hence, a new folder is created at the path provided.'
        )
        os.makedirs(target_folder, exist_ok=True)
```

**Review by mostaphaRoudsari — APPROVED** [review]
Hi @devngc, see my comment! Looks good otherwise. Feel free to merge it in after addressing the comment. Thanks.

**Pull request merged**